tmt/steps/report/reportportal.py

```diff
@@ -143,8 +153,82 @@ def go(self) -> None:
             message = response.text
         if not response.ok:
             raise tmt.utils.ReportError(
-                f"Received non-ok status code from ReportPortal, response text is: {message}")
+                "Received non-ok status code from ReportPortal, "
+                f"response text is: {message}")
         else:
             self.debug(f"Response code from the server: {response.status_code}")
             self.debug(f"Message from the server: {message}")
             self.info("report", "Successfully uploaded.", "yellow")
+
+        try:
+            _ = json.dumps(meta_data)
+        except Exception as exc:
+            raise tmt.utils.SpecificationError("Invalid syntax of provided metadata") from exc
```

Comment:
I would move this test under `if meta_data` condition below - after all, if `meta_data` is not set, there's no need to verify it's a valid JSON since it's not going to be used anyway.

Reply:
Ack